File: app/src/main/java/com/example/amit/uniconnexample/Others/UserData.java
package com.example.amit.uniconnexample.Others;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by amit on 30/10/16.
 */

public class UserData implements Parcelable {
    public String phone,  name;
    public String email , photo ,clg; //extras


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.phone);
        dest.writeString(this.name);
      //  dest.writeString(this.location);
        dest.writeString(this.photo);
        dest.writeString(this.clg);
    }

    public UserData() {
    }

    protected UserData(Parcel in) {
        this.phone = in.readString();
        this.name = in.readString();
      //  this.location = in.readString();
        this.photo = in.readString();
        this.clg=in.readString();
    }

    public static final Creator<UserData> CREATOR = new Parcelable.Creator<UserData>() {
        @Override
        public UserData createFromParcel(Parcel source) {
            return new UserData(source);
        }

        @Override
        public UserData[] newArray(int size) {
            return new UserData[size];
        }
    };
}

File: app/src/main/java/com/example/amit/uniconnexample/Others/UserData.kt
package com.example.amit.uniconnexample.Others

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by amit on 30/10/16.
 */

class UserData : Parcelable {
    var phone: String? = null
    var name: String? = null
    var email: String? = null
    var photo: String? = null
    var clg: String? = null //extras


    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.phone)
        dest.writeString(this.name)
        //  dest.writeString(this.location);
        dest.writeString(this.photo)
        dest.writeString(this.clg)
    }

    constructor() {}

    protected constructor(`in`: Parcel) {
        this.phone = `in`.readString()
        this.name = `in`.readString()
        //  this.location = in.readString();
        this.photo = `in`.readString()
        this.clg = `in`.readString()
    }

    companion object {

        @JvmField val CREATOR: Parcelable.Creator<UserData> = object : Parcelable.Creator<UserData> {
            override fun createFromParcel(source: Parcel): UserData {
                return UserData(source)
            }

            override fun newArray(size: Int): Array<UserData?> {
                return arrayOfNulls(size)
            }
        }
    }
}